include/aspect/material_model/utilities.h

```diff
@@ -549,6 +549,187 @@ namespace aspect
+
+          /**
+           * Return how many phases there are for each composition.
+           * Note, that most likely you only need the number of phase transitions
+           * for each chemical composition, so use the function above instead.
+           * This function is only kept for backward compatibility.
+           */
+          const std::vector<unsigned int> &
+          n_phases_for_each_composition () const;
+
+          /**
+           * Declare the parameters this class takes through input files.
+           * Note that this class does not declare its own subsection,
+           * i.e. the parameters will be declared in the subsection that
+           * was active before calling this function.
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters this class declares from the parameter file.
+           * Note that this class does not declare its own subsection,
+           * i.e. the parameters will be parsed from the subsection that
+           * was active before calling this function.
+           */
+          void
+          parse_parameters (ParameterHandler &prm);
+
+
+        private:
+          /**
+          * Directory path where data files are stored.
+          */
+          std::string data_directory;
+
+          /**
+           * List of file names containing material data for each composition.
+          */
+          std::vector<std::string> material_file_names;
+
+          /**
+          * Minimum temperature values for each composition in the P-T table.
+          */
+          std::vector<double> minimum_temperature;
+
+          /**
+          * Maximum temperature values for each composition in the P-T table.
+          */
+          std::vector<double> maximum_temperature;
+
+          /**
+          * Temperature intervals used for each composition in the P-T table.
+          */
+          std::vector<double> interval_temperature;
+
+          /**
+          * Minimum pressure values for each composition in the P-T table.
+          */
+          std::vector<double> minimum_pressure;
+
+          /**
+          * Maximum pressure values for each composition in the P-T table.
+          */
+          std::vector<double> maximum_pressure;
+
+          /**
+          * Pressure intervals used for each composition in the P-T table.
+          */
+          std::vector<double> interval_pressure;
+
+          /**
+           * List of pointers to objects that read and process data we get from
+           * material data files. There is one pointer/object per lookup file.
+           */
+          std::vector<std::unique_ptr<Utilities::StructuredDataLookup<2>>> material_lookup;
+
+          /**
+           * List of phase indicators of the most dominant phases in the material data files
+           * to construct the different phase transitions in this class
```

Comment:
Theoretically you can make this work with either `double` or `unsigned int` as long as you always convert correctly. But converting is always error-prone, and `unsigned int` is just the more natural data type for a variable that represents an integer number (i.e. a quantity that has discrete and separate values).

I would do the following:
- use `unsigned int` for the dominant phase index and the data from the table and prm file everywhere
- the only place where you have to convert from double to int should then be [here](https://github.com/geodynamics/aspect/pull/6131/files#diff-252e9f5000481fac6894687cc01e92993cd5e215dba9f4b67c5ab331e2eb9abbR1234), because the `StructuredData` class unfortunately interprets all the data columns as if they are `double`.

Reply:
Yes, I have checked this is how the implementation currently works